From b19a47cde4cb0627524ad48f114b39dd687a8dff Mon Sep 17 00:00:00 2001
From: The jR3f <97492766+jR3f@users.noreply.github.com>
Date: Mon, 20 Dec 2021 18:59:18 +0800
Subject: [PATCH] Translate Biome Names Before 1.18 to zh-CHS


diff --git a/util.c b/util.c
index e327664..5168f59 100644
--- a/util.c
+++ b/util.c
@@ -97,35 +97,35 @@ const char *biome2str(int mc, int id)
     case frozen_ocean: return "frozen_ocean";
     case frozen_river: return "frozen_river";
     case snowy_tundra: return "snowy_tundra";
-    case snowy_mountains: return "snowy_mountains";
+    case snowy_mountains: return "雪山";
     case mushroom_fields: return "mushroom_fields";
-    case mushroom_field_shore: return "mushroom_field_shore";
+    case mushroom_field_shore: return "蘑菇岛岸";
     case beach: return "beach";
-    case desert_hills: return "desert_hills";
-    case wooded_hills: return "wooded_hills";
-    case taiga_hills: return "taiga_hills";
+    case desert_hills: return "沙漠丘陵";
+    case wooded_hills: return "繁茂的丘陵";
+    case taiga_hills: return "针叶林丘陵";
     // 20
-    case mountain_edge: return "mountain_edge";
+    case mountain_edge: return "山地边缘";
     case jungle: return "jungle";
-    case jungle_hills: return "jungle_hills";
+    case jungle_hills: return "丛林丘陵";
     case jungle_edge: return "jungle_edge";
     case deep_ocean: return "deep_ocean";
     case stone_shore: return "stone_shore";
     case snowy_beach: return "snowy_beach";
     case birch_forest: return "birch_forest";
-    case birch_forest_hills: return "birch_forest_hills";
+    case birch_forest_hills: return "桦木森林丘陵";
     case dark_forest: return "dark_forest";
     // 30
     case snowy_taiga: return "snowy_taiga";
-    case snowy_taiga_hills: return "snowy_taiga_hills";
+    case snowy_taiga_hills: return "积雪的针叶林丘陵";
     case giant_tree_taiga: return "giant_tree_taiga";
-    case giant_tree_taiga_hills: return "giant_tree_taiga_hills";
+    case giant_tree_taiga_hills: return "巨型针叶林丘陵";
     case wooded_mountains: return "wooded_mountains";
     case savanna: return "savanna";
     case savanna_plateau: return "savanna_plateau";
     case badlands: return "badlands";
     case wooded_badlands_plateau: return "wooded_badlands_plateau";
-    case badlands_plateau: return "badlands_plateau";
+    case badlands_plateau: return "恶地高原";
     // 40  --  1.13
     case small_end_islands: return "small_end_islands";
     case end_midlands: return "end_midlands";
@@ -134,7 +134,7 @@ const char *biome2str(int mc, int id)
     case warm_ocean: return "warm_ocean";
     case lukewarm_ocean: return "lukewarm_ocean";
     case cold_ocean: return "cold_ocean";
-    case deep_warm_ocean: return "deep_warm_ocean";
+    case deep_warm_ocean: return "暖水深海";
     case deep_lukewarm_ocean: return "deep_lukewarm_ocean";
     case deep_cold_ocean: return "deep_cold_ocean";
     // 50
@@ -144,29 +144,29 @@ const char *biome2str(int mc, int id)
 
     // mutated variants
     case sunflower_plains: return "sunflower_plains";
-    case desert_lakes: return "desert_lakes";
+    case desert_lakes: return "沙漠湖泊";
     case gravelly_mountains: return "gravelly_mountains";
     case flower_forest: return "flower_forest";
-    case taiga_mountains: return "taiga_mountains";
-    case swamp_hills: return "swamp_hills";
+    case taiga_mountains: return "针叶林山地";
+    case swamp_hills: return "沼泽丘陵";
     case ice_spikes: return "ice_spikes";
-    case modified_jungle: return "modified_jungle";
-    case modified_jungle_edge: return "modified_jungle_edge";
+    case modified_jungle: return "丛林变种";
+    case modified_jungle_edge: return "丛林边缘变种";
     case tall_birch_forest: return "tall_birch_forest";
-    case tall_birch_hills: return "tall_birch_hills";
-    case dark_forest_hills: return "dark_forest_hills";
-    case snowy_taiga_mountains: return "snowy_taiga_mountains";
+    case tall_birch_hills: return "高大桦木丘陵";
+    case dark_forest_hills: return "黑森林丘陵";
+    case snowy_taiga_mountains: return "积雪的针叶林山地";
     case giant_spruce_taiga: return "giant_spruce_taiga";
-    case giant_spruce_taiga_hills: return "giant_spruce_taiga_hills";
-    case modified_gravelly_mountains: return "modified_gravelly_mountains";
+    case giant_spruce_taiga_hills: return "巨型云杉针叶林丘陵";
+    case modified_gravelly_mountains: return "沙砾山地+";
     case shattered_savanna: return "shattered_savanna";
-    case shattered_savanna_plateau: return "shattered_savanna_plateau";
+    case shattered_savanna_plateau: return "破碎的热带高原";
     case eroded_badlands: return "eroded_badlands";
-    case modified_wooded_badlands_plateau: return "modified_wooded_badlands_plateau";
-    case modified_badlands_plateau: return "modified_badlands_plateau";
+    case modified_wooded_badlands_plateau: return "繁茂的恶地高原变种";
+    case modified_badlands_plateau: return "恶地高原变种";
     // 1.14
     case bamboo_jungle: return "bamboo_jungle";
-    case bamboo_jungle_hills: return "bamboo_jungle_hills";
+    case bamboo_jungle_hills: return "竹林丘陵";
     // 1.16
     case soul_sand_valley: return "soul_sand_valley";
     case crimson_forest: return "crimson_forest";
-- 
2.34.1

