From b40b505ce27192c8da9cc8dfebe9e9723fa6cc57 Mon Sep 17 00:00:00 2001
From: The jR3f <97492766+jR3f@users.noreply.github.com>
Date: Mon, 20 Dec 2021 19:24:54 +0800
Subject: [PATCH] Translate Biome Names in 1.18 to zh-CHS


diff --git a/util.c b/util.c
index 5168f59..ce68eeb 100644
--- a/util.c
+++ b/util.c
@@ -67,89 +67,89 @@ const char *biome2str(int mc, int id)
         // (based on their features and biome id conversion when upgrading)
         switch (id)
         {
-        case old_growth_birch_forest: return "old_growth_birch_forest";
-        case old_growth_pine_taiga: return "old_growth_pine_taiga";
-        case old_growth_spruce_taiga: return "old_growth_spruce_taiga";
-        case snowy_plains: return "snowy_plains";
-        case sparse_jungle: return "sparse_jungle";
-        case stony_shore: return "stony_shore";
-        case windswept_hills: return "windswept_hills";
-        case windswept_forest: return "windswept_forest";
-        case windswept_gravelly_hills: return "windswept_gravelly_hills";
-        case windswept_savanna: return "windswept_savanna";
-        case wooded_badlands: return "wooded_badlands";
+        case old_growth_birch_forest: return "原始桦木森林";
+        case old_growth_pine_taiga: return "原始松木针叶林";
+        case old_growth_spruce_taiga: return "原始云杉针叶林";
+        case snowy_plains: return "积雪的平原";
+        case sparse_jungle: return "稀疏的丛林";
+        case stony_shore: return "石岸";
+        case windswept_hills: return "风袭丘陵";
+        case windswept_forest: return "风袭森林";
+        case windswept_gravelly_hills: return "风袭沙砾丘陵";
+        case windswept_savanna: return "风袭热带草原";
+        case wooded_badlands: return "繁茂的恶地";
         }
     }
 
     switch (id)
     {
-    case ocean: return "ocean";
-    case plains: return "plains";
-    case desert: return "desert";
+    case ocean: return "海洋";
+    case plains: return "平原";
+    case desert: return "沙漠";
     case mountains: return "mountains";
-    case forest: return "forest";
-    case taiga: return "taiga";
-    case swamp: return "swamp";
-    case river: return "river";
-    case nether_wastes: return "nether_wastes";
-    case the_end: return "the_end";
+    case forest: return "森林";
+    case taiga: return "针叶林";
+    case swamp: return "沼泽";
+    case river: return "河流";
+    case nether_wastes: return "下界荒地";
+    case the_end: return "末地";
     // 10
-    case frozen_ocean: return "frozen_ocean";
-    case frozen_river: return "frozen_river";
+    case frozen_ocean: return "冻洋";
+    case frozen_river: return "冻河";
     case snowy_tundra: return "snowy_tundra";
     case snowy_mountains: return "雪山";
-    case mushroom_fields: return "mushroom_fields";
+    case mushroom_fields: return "蘑菇岛";
     case mushroom_field_shore: return "蘑菇岛岸";
-    case beach: return "beach";
+    case beach: return "沙滩";
     case desert_hills: return "沙漠丘陵";
     case wooded_hills: return "繁茂的丘陵";
     case taiga_hills: return "针叶林丘陵";
     // 20
     case mountain_edge: return "山地边缘";
-    case jungle: return "jungle";
+    case jungle: return "丛林";
     case jungle_hills: return "丛林丘陵";
     case jungle_edge: return "jungle_edge";
-    case deep_ocean: return "deep_ocean";
+    case deep_ocean: return "深海";
     case stone_shore: return "stone_shore";
-    case snowy_beach: return "snowy_beach";
-    case birch_forest: return "birch_forest";
+    case snowy_beach: return "积雪的沙滩";
+    case birch_forest: return "桦木森林";
     case birch_forest_hills: return "桦木森林丘陵";
-    case dark_forest: return "dark_forest";
+    case dark_forest: return "黑森林";
     // 30
-    case snowy_taiga: return "snowy_taiga";
+    case snowy_taiga: return "积雪的针叶林";
     case snowy_taiga_hills: return "积雪的针叶林丘陵";
     case giant_tree_taiga: return "giant_tree_taiga";
     case giant_tree_taiga_hills: return "巨型针叶林丘陵";
     case wooded_mountains: return "wooded_mountains";
-    case savanna: return "savanna";
-    case savanna_plateau: return "savanna_plateau";
-    case badlands: return "badlands";
+    case savanna: return "热带草原";
+    case savanna_plateau: return "热带高原";
+    case badlands: return "恶地";
     case wooded_badlands_plateau: return "wooded_badlands_plateau";
     case badlands_plateau: return "恶地高原";
     // 40  --  1.13
-    case small_end_islands: return "small_end_islands";
-    case end_midlands: return "end_midlands";
-    case end_highlands: return "end_highlands";
-    case end_barrens: return "end_barrens";
-    case warm_ocean: return "warm_ocean";
-    case lukewarm_ocean: return "lukewarm_ocean";
-    case cold_ocean: return "cold_ocean";
+    case small_end_islands: return "末地小型岛屿";
+    case end_midlands: return "末地中型岛屿";
+    case end_highlands: return "末地高岛";
+    case end_barrens: return "末地荒岛";
+    case warm_ocean: return "暖水海洋";
+    case lukewarm_ocean: return "温水海洋";
+    case cold_ocean: return "冷水海洋";
     case deep_warm_ocean: return "暖水深海";
-    case deep_lukewarm_ocean: return "deep_lukewarm_ocean";
-    case deep_cold_ocean: return "deep_cold_ocean";
+    case deep_lukewarm_ocean: return "温水深海";
+    case deep_cold_ocean: return "冷水深海";
     // 50
-    case deep_frozen_ocean: return "deep_frozen_ocean";
+    case deep_frozen_ocean: return "冰冻深海";
 
-    case the_void: return "the_void";
+    case the_void: return "虚空";
 
     // mutated variants
-    case sunflower_plains: return "sunflower_plains";
+    case sunflower_plains: return "向日葵平原";
     case desert_lakes: return "沙漠湖泊";
     case gravelly_mountains: return "gravelly_mountains";
-    case flower_forest: return "flower_forest";
+    case flower_forest: return "繁花森林";
     case taiga_mountains: return "针叶林山地";
     case swamp_hills: return "沼泽丘陵";
-    case ice_spikes: return "ice_spikes";
+    case ice_spikes: return "冰刺平原";
     case modified_jungle: return "丛林变种";
     case modified_jungle_edge: return "丛林边缘变种";
     case tall_birch_forest: return "tall_birch_forest";
@@ -161,27 +161,27 @@ const char *biome2str(int mc, int id)
     case modified_gravelly_mountains: return "沙砾山地+";
     case shattered_savanna: return "shattered_savanna";
     case shattered_savanna_plateau: return "破碎的热带高原";
-    case eroded_badlands: return "eroded_badlands";
+    case eroded_badlands: return "被风蚀的恶地";
     case modified_wooded_badlands_plateau: return "繁茂的恶地高原变种";
     case modified_badlands_plateau: return "恶地高原变种";
     // 1.14
-    case bamboo_jungle: return "bamboo_jungle";
+    case bamboo_jungle: return "竹林";
     case bamboo_jungle_hills: return "竹林丘陵";
     // 1.16
-    case soul_sand_valley: return "soul_sand_valley";
-    case crimson_forest: return "crimson_forest";
-    case warped_forest: return "warped_forest";
-    case basalt_deltas: return "basalt_deltas";
+    case soul_sand_valley: return "灵魂沙峡谷";
+    case crimson_forest: return "绯红森林";
+    case warped_forest: return "诡异森林";
+    case basalt_deltas: return "玄武岩三角洲";
     // 1.17
-    case dripstone_caves: return "dripstone_caves";
-    case lush_caves: return "lush_caves";
+    case dripstone_caves: return "溶洞";
+    case lush_caves: return "繁茂洞穴";
     // 1.18
-    case meadow: return "meadow";
-    case grove: return "grove";
-    case snowy_slopes: return "snowy_slopes";
-    case stony_peaks: return "stony_peaks";
-    case jagged_peaks: return "jagged_peaks";
-    case frozen_peaks: return "frozen_peaks";
+    case meadow: return "草甸";
+    case grove: return "雪林";
+    case snowy_slopes: return "积雪的山坡";
+    case stony_peaks: return "裸岩山峰";
+    case jagged_peaks: return "尖峭山峰";
+    case frozen_peaks: return "冰封山峰";
     }
     return NULL;
 }
-- 
2.34.1

