From f30d38a501771ba9a6cadf99e7e7a3032379cd27 Mon Sep 17 00:00:00 2001
From: The jR3f <97492766+jR3f@users.noreply.github.com>
Date: Tue, 21 Dec 2021 00:45:47 +0800
Subject: [PATCH] Translate FormSearchControl to zh-CHS


diff --git a/src/formsearchcontrol.cpp b/src/formsearchcontrol.cpp
index 1acdc66..ff69f43 100644
--- a/src/formsearchcontrol.cpp
+++ b/src/formsearchcontrol.cpp
@@ -148,7 +148,7 @@ void FormSearchControl::searchLockUi(bool lock)
     }
     else
     {
-        ui->buttonStart->setText("Start search");
+        ui->buttonStart->setText("开始搜索");
         ui->buttonStart->setIcon(QIcon(":/icons/search.png"));
         ui->buttonStart->setChecked(false);
         ui->buttonStart->setEnabled(true);
@@ -227,7 +227,7 @@ void FormSearchControl::on_buttonStart_clicked()
         if (ok)
         {
             ui->lineStart->setText(QString::asprintf("%" PRId64, (int64_t)sc.startseed));
-            ui->buttonStart->setText("Abort search");
+            ui->buttonStart->setText("中止搜索");
             ui->buttonStart->setIcon(QIcon(":/icons/cancel.png"));
             sthread.start();
             searchLockUi(true);
@@ -242,7 +242,7 @@ void FormSearchControl::on_buttonStart_clicked()
         sthread.stop(); // tell search to stop at next convenience
         //sthread.quit(); // tell the event loop to exit
         //sthread.wait(); // wait for search to finish
-        ui->buttonStart->setText("Start search");
+        ui->buttonStart->setText("开始搜索");
         ui->buttonStart->setIcon(QIcon(":/icons/search.png"));
         ui->buttonStart->setChecked(false);
 
diff --git a/src/formsearchcontrol.ui b/src/formsearchcontrol.ui
index d9e70fc..f77fadb 100644
--- a/src/formsearchcontrol.ui
+++ b/src/formsearchcontrol.ui
@@ -83,7 +83,7 @@
      </attribute>
      <column>
       <property name="text">
-       <string>Hex  (Low-48 | Top-16)</string>
+       <string>16进制 (低48位|高16位)</string>
       </property>
       <property name="textAlignment">
        <set>AlignLeading|AlignVCenter</set>
@@ -91,7 +91,7 @@
      </column>
      <column>
       <property name="text">
-       <string>Seed</string>
+       <string>种子</string>
       </property>
       <property name="textAlignment">
        <set>AlignLeading|AlignVCenter</set>
@@ -133,7 +133,7 @@
      <item row="0" column="5">
       <widget class="QLabel" name="label_3">
        <property name="text">
-        <string>Threads:</string>
+        <string>线程数:</string>
        </property>
       </widget>
      </item>
@@ -143,7 +143,7 @@
         <string>stop as soon as the next set of matching seeds is found</string>
        </property>
        <property name="text">
-        <string>Stop on results</string>
+        <string>在搜索到结果时停止</string>
        </property>
        <property name="checked">
         <bool>true</bool>
@@ -156,7 +156,7 @@
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;searching algorithm&lt;/p&gt;&lt;p&gt;incremental: best for non-exhaustive biome searches&lt;/p&gt;&lt;p&gt;48-bit family blocks: best for searches with strong structure restrictions&lt;/p&gt;&lt;p&gt;seed list from file...: load and search through a pre-computed list of seed&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
-        <string>Search:</string>
+        <string>搜索类型:</string>
        </property>
       </widget>
      </item>
@@ -189,7 +189,7 @@
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
-        <string>Seed:</string>
+        <string>当前种子:</string>
        </property>
       </widget>
      </item>
@@ -213,17 +213,17 @@
        </property>
        <item>
         <property name="text">
-         <string>incremental</string>
+         <string>自增遍历</string>
         </property>
        </item>
        <item>
         <property name="text">
-         <string>48-bit family blocks</string>
+         <string>低48位簇块筛选</string>
         </property>
        </item>
        <item>
         <property name="text">
-         <string>seed list from a file...</string>
+         <string>从文件获取种子列表...</string>
         </property>
        </item>
       </widget>
@@ -257,7 +257,7 @@
        <item>
         <widget class="QPushButton" name="buttonClear">
          <property name="text">
-          <string>Clear results</string>
+          <string>清除结果</string>
          </property>
          <property name="icon">
           <iconset resource="../icons.qrc">
@@ -268,7 +268,7 @@
        <item>
         <widget class="QPushButton" name="buttonStart">
          <property name="text">
-          <string>Start search</string>
+          <string>开始搜索</string>
          </property>
          <property name="icon">
           <iconset resource="../icons.qrc">
-- 
2.34.1

