From e1713b776aaae32dfbc275e9d8ea6aab4e12a569 Mon Sep 17 00:00:00 2001
From: The jR3f <97492766+jR3f@users.noreply.github.com>
Date: Tue, 11 Jan 2022 15:40:59 +0800
Subject: [PATCH] Translate FormGen48 to zh-CHS


diff --git a/src/formgen48.cpp b/src/formgen48.cpp
index 0055cf7..f3ea454 100644
--- a/src/formgen48.cpp
+++ b/src/formgen48.cpp
@@ -133,15 +133,15 @@ bool FormGen48::setList48(QString path, bool quiet)
 
     if (slist48path.isEmpty())
     {
-        ui->lineList48->setText(tr("[none]"));
+        ui->lineList48->setText(tr("[空]"));
     }
     else
     {
         QString fname = QFileInfo(path).baseName();
         if (slist48.empty())
-            ui->lineList48->setText(tr("[no seeds!] %1").arg(fname));
+            ui->lineList48->setText(tr("[没有种子!] %1").arg(fname));
         else
-            ui->lineList48->setText(tr("[%n seed(s)] %1", "", slist48.size()).arg(fname));
+            ui->lineList48->setText(tr("[%n 个种子] %1", "", slist48.size()).arg(fname));
     }
 
     emit changed();
@@ -255,7 +255,7 @@ void FormGen48::updateCount()
 
     if (cnt >= MASK48+1)
     {
-        ui->labelCount->setText(tr("all", "Checking all 64-bit seeds"));
+        ui->labelCount->setText(tr("全部", "Checking all 64-bit seeds"));
     }
     else
     {
@@ -286,11 +286,11 @@ void FormGen48::updateAutoUi()
     bool isqh = cond.type >= F_QH_IDEAL && cond.type <= F_QH_BARELY;
     bool isqm = cond.type >= F_QM_95 && cond.type <= F_QM_90;
     if (isqh)
-        modestr = tr("[Quad-hut]");
+        modestr = tr("[四联女巫小屋]");
     else if (isqm)
-        modestr = tr("[Quad-monument]");
+        modestr = tr("[四联海底神殿]");
     else
-        modestr = tr("[None]");
+        modestr = tr("[空]");
 
     ui->labelAuto->setText(modestr);
 
diff --git a/src/formgen48.ui b/src/formgen48.ui
index a6dc3bc..26548c5 100644
--- a/src/formgen48.ui
+++ b/src/formgen48.ui
@@ -36,7 +36,7 @@
         </property>
         <widget class="QWidget" name="tabAuto">
          <attribute name="title">
-          <string>Auto/None</string>
+          <string>自动/空</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_4">
           <property name="margin">
@@ -45,7 +45,7 @@
           <item row="0" column="0">
            <widget class="QLabel" name="labelAutoDesc">
             <property name="text">
-             <string>Automatically select suitable options based on conditions.</string>
+             <string>自动基于搜索条件选择合适的选项</string>
             </property>
            </widget>
           </item>
@@ -60,7 +60,7 @@
         </widget>
         <widget class="QWidget" name="tabQuadF">
          <attribute name="title">
-          <string>Quad-feature</string>
+          <string>四联结构</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_5">
           <property name="margin">
@@ -72,7 +72,7 @@
              <string>Value (such as a structure salt) added to the 48-bit seeds</string>
             </property>
             <property name="text">
-             <string>Add structure salt:</string>
+             <string>添加结构盐值:</string>
             </property>
            </widget>
           </item>
@@ -93,7 +93,7 @@
 Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in particular swamp-huts</string>
             </property>
             <property name="text">
-             <string>Restrict lower bits:</string>
+             <string>限制低比特位:</string>
             </property>
            </widget>
           </item>
@@ -104,27 +104,27 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
             </property>
             <item>
              <property name="text">
-              <string>Quad-hut (ideal)</string>
+              <string>四联女巫小屋 (理想)</string>
              </property>
             </item>
             <item>
              <property name="text">
-              <string>Quad-hut (classic)</string>
+              <string>四联女巫小屋 (经典)</string>
              </property>
             </item>
             <item>
              <property name="text">
-              <string>Quad-hut (normal)</string>
+              <string>四联女巫小屋 (普通)</string>
              </property>
             </item>
             <item>
              <property name="text">
-              <string>Quad-hut (barely)</string>
+              <string>四联女巫小屋 (稀疏)</string>
              </property>
             </item>
             <item>
              <property name="text">
-              <string>Quad-feature (ideal, custom salt)</string>
+              <string>四联结构特征 (理想, 自定义盐值)</string>
              </property>
             </item>
            </widget>
@@ -133,7 +133,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
         </widget>
         <widget class="QWidget" name="tabQuadM">
          <attribute name="title">
-          <string>Quad-monument</string>
+          <string>四联海底神殿</string>
          </attribute>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
@@ -143,7 +143,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
 (i.e. this is a measure quality, out of 4*58*58)</string>
             </property>
             <property name="text">
-             <string>Min. monument area:</string>
+             <string>最小海底神殿区域:</string>
             </property>
            </widget>
           </item>
@@ -151,7 +151,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
         </widget>
         <widget class="QWidget" name="tabList">
          <attribute name="title">
-          <string>Seed list</string>
+          <string>种子列表</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_7">
           <property name="margin">
@@ -163,7 +163,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
              <string>Use a custom seed list as source for 48-bit seed candidates</string>
             </property>
             <property name="text">
-             <string>Load 48-bit seed list:</string>
+             <string>加载48位种子列表:</string>
             </property>
            </widget>
           </item>
@@ -199,7 +199,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
              <string>Value (such as a structure salt) added to the 48-bit seeds</string>
             </property>
             <property name="text">
-             <string>Add structure salt:</string>
+             <string>添加结构盐值:</string>
             </property>
            </widget>
           </item>
@@ -248,21 +248,21 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
          <item row="0" column="2">
           <widget class="QRadioButton" name="radioAuto">
            <property name="text">
-            <string>Auto</string>
+            <string>自动</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QRadioButton" name="radioManual">
            <property name="text">
-            <string>Manual</string>
+            <string>手动</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="labelTranspose">
            <property name="text">
-            <string>Transpose by regions (multiplied by ×512):</string>
+            <string>按区域转置 (倍率x512):</string>
            </property>
           </widget>
          </item>
@@ -274,7 +274,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
         <item>
          <widget class="QLabel" name="labelX1">
           <property name="toolTip">
-           <string>Lower bound (inclusive)</string>
+           <string>最小值 (包含)</string>
           </property>
           <property name="text">
            <string>X&lt;sub&gt;1&lt;/sub&gt;=</string>
@@ -284,7 +284,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
         <item>
          <widget class="QLineEdit" name="lineEditX1">
           <property name="toolTip">
-           <string>Lower bound (inclusive)</string>
+           <string>最小值 (包含)</string>
           </property>
           <property name="text">
            <string>0</string>
@@ -294,7 +294,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
         <item>
          <widget class="QLabel" name="labelZ1">
           <property name="toolTip">
-           <string>Lower bound (inclusive)</string>
+           <string>最小值 (包含)</string>
           </property>
           <property name="text">
            <string>Z&lt;sub&gt;1&lt;/sub&gt;=</string>
@@ -304,7 +304,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
         <item>
          <widget class="QLineEdit" name="lineEditZ1">
           <property name="toolTip">
-           <string>Lower bound (inclusive)</string>
+           <string>最小值 (包含)</string>
           </property>
           <property name="text">
            <string>0</string>
@@ -327,7 +327,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
         <item>
          <widget class="QLabel" name="labelX2">
           <property name="toolTip">
-           <string>Upper bound (inclusive)</string>
+           <string>最大值 (包含)</string>
           </property>
           <property name="text">
            <string>X&lt;sub&gt;2&lt;/sub&gt;=</string>
@@ -337,7 +337,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
         <item>
          <widget class="QLineEdit" name="lineEditX2">
           <property name="toolTip">
-           <string>Upper bound (inclusive)</string>
+           <string>最大值 (包含)</string>
           </property>
           <property name="text">
            <string>0</string>
@@ -347,7 +347,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
         <item>
          <widget class="QLabel" name="labelZ2">
           <property name="toolTip">
-           <string>Upper bound (inclusive)</string>
+           <string>最大值 (包含)</string>
           </property>
           <property name="text">
            <string>Z&lt;sub&gt;2&lt;/sub&gt;=</string>
@@ -357,7 +357,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
         <item>
          <widget class="QLineEdit" name="lineEditZ2">
           <property name="toolTip">
-           <string>Upper bound (inclusive)</string>
+           <string>最大值 (包含)</string>
           </property>
           <property name="text">
            <string>0</string>
@@ -369,7 +369,7 @@ Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in par
       <item row="3" column="0">
        <widget class="QLabel" name="labelSpace">
         <property name="text">
-         <string>Seeds in search space:</string>
+         <string>搜索区域内的种子数:</string>
         </property>
        </widget>
       </item>
-- 
2.35.1

