From 176895f37622b42d0edb6370ee98bcbeec969d51 Mon Sep 17 00:00:00 2001
From: The jR3f <97492766+jR3f@users.noreply.github.com>
Date: Tue, 11 Jan 2022 16:14:08 +0800
Subject: [PATCH] Translate FilterDialog to zh-CHS


diff --git a/src/filterdialog.cpp b/src/filterdialog.cpp
index 3d9780d..0617a70 100644
--- a/src/filterdialog.cpp
+++ b/src/filterdialog.cpp
@@ -63,10 +63,10 @@ FilterDialog::FilterDialog(FormConditions *parent, Config *config, int mcversion
 
     textDescription = new QTextEdit(this);
     textDescription->setSizeAdjustPolicy(QAbstractScrollArea::AdjustToContents);
-    ui->collapseDescription->init(tr("Description"), textDescription, true);
+    ui->collapseDescription->init(tr("介绍"), textDescription, true);
 
     const char *p_mcs = mc2str(mc);
-    QString mcs = tr("MC %1", "Minecraft version").arg(p_mcs ? p_mcs : "?");
+    QString mcs = tr("Minecraft %1", "Minecraft version").arg(p_mcs ? p_mcs : "?");
     ui->labelMC->setText(mcs);
 
 
@@ -218,22 +218,22 @@ FilterDialog::FilterDialog(FormConditions *parent, Config *config, int mcversion
     SETUP_TEMPCAT_SPINBOX(Special+Lush);
     SETUP_TEMPCAT_SPINBOX(Special+Cold);
 
-    addVariant("plains_fountain_01", plains, 0);
-    addVariant("plains_meeting_point_1", plains, 1);
-    addVariant("plains_meeting_point_2", plains, 2);
-    addVariant("plains_meeting_point_3", plains, 3);
-    addVariant("desert_meeting_point_1", desert, 1);
-    addVariant("desert_meeting_point_2", desert, 2);
-    addVariant("desert_meeting_point_3", desert, 3);
-    addVariant("savanna_meeting_point_1", savanna, 1);
-    addVariant("savanna_meeting_point_2", savanna, 2);
-    addVariant("savanna_meeting_point_3", savanna, 3);
-    addVariant("savanna_meeting_point_4", savanna, 4);
-    addVariant("taiga_meeting_point_1", taiga, 1);
-    addVariant("taiga_meeting_point_2", taiga, 2);
-    addVariant("snowy_meeting_point_1", snowy_tundra, 1);
-    addVariant("snowy_meeting_point_2", snowy_tundra, 2);
-    addVariant("snowy_meeting_point_3", snowy_tundra, 3);
+    addVariant("平原喷泉 0", plains, 0);
+    addVariant("平原交汇点 1", plains, 1);
+    addVariant("平原交汇点 2", plains, 2);
+    addVariant("平原交汇点 3", plains, 3);
+    addVariant("沙漠交汇点 1", desert, 1);
+    addVariant("沙漠交汇点 2", desert, 2);
+    addVariant("沙漠交汇点 3", desert, 3);
+    addVariant("热带草原交汇点 1", savanna, 1);
+    addVariant("热带草原交汇点 2", savanna, 2);
+    addVariant("热带草原交汇点 3", savanna, 3);
+    addVariant("热带草原交汇点 4", savanna, 4);
+    addVariant("针叶林交汇点 1", taiga, 1);
+    addVariant("针叶林交汇点 2", taiga, 2);
+    addVariant("雪原交汇点 1", snowy_tundra, 1);
+    addVariant("雪原交汇点 2", snowy_tundra, 2);
+    addVariant("雪原交汇点 3", snowy_tundra, 3);
 
     ui->scrollBiomes->setStyleSheet(
             "QCheckBox::indicator:unchecked     { image: url(:/icons/check0.png); }\n"
@@ -467,17 +467,17 @@ void FilterDialog::updateMode()
     if (ft.step > 1)
     {
         QString multxt = QString("%1%2").arg(QChar(0xD7)).arg(ft.step);
-        loc = tr("Location (coordinates are multiplied by %1)").arg(multxt);
-        areatip = tr("From floor(-x/2)%1 to floor(x/2)%1 on both axes (inclusive)").arg(multxt);
-        lowtip = tr("Lower bound %1 (inclusive)").arg(multxt);
-        uptip = tr("Upper bound %1 (inclusive)").arg(multxt);
+        loc = tr("位置 (坐标倍率 %1)").arg(multxt);
+        areatip = tr("From floor(-x/2)%1 to floor(x/2)%1 on both axes (包含)").arg(multxt);
+        lowtip = tr("最小值 %1 (包含)").arg(multxt);
+        uptip = tr("最大值 %1 (包含)").arg(multxt);
     }
     else
     {
-        loc = tr("Location");
-        areatip = tr("From floor(-x/2) to floor(x/2) on both axes (inclusive)");
-        lowtip = tr("Lower bound (inclusive)");
-        uptip = tr("Upper bound (inclusive)");
+        loc = tr("位置");
+        areatip = tr("From floor(-x/2) to floor(x/2) on both axes (包含)");
+        lowtip = tr("最小值 (包含)");
+        uptip = tr("最大值 (包含)");
     }
     ui->groupBoxPosition->setTitle(loc);
     ui->radioSquare->setToolTip(areatip);
@@ -804,7 +804,7 @@ void FilterDialog::on_comboBoxCat_currentIndexChanged(int idx)
     ui->comboBoxType->clear();
 
     int slot = 0;
-    ui->comboBoxType->insertItem(slot, tr("Select filter"), QVariant::fromValue((int)F_SELECT));
+    ui->comboBoxType->insertItem(slot, tr("选择过滤器"), QVariant::fromValue((int)F_SELECT));
 
     const FilterInfo *ft_list[FILTER_MAX] = {};
     const FilterInfo *ft;
diff --git a/src/filterdialog.h b/src/filterdialog.h
index d1d07ea..2b9ee0b 100644
--- a/src/filterdialog.h
+++ b/src/filterdialog.h
@@ -36,9 +36,9 @@ public:
     {
         QString txt = QSpinBox::textFromValue(value);
         if (value == 0)
-            txt += " " + tr("(ignore)");
+            txt += " " + tr("(忽略)");
         if (value == -1)
-            txt += " " + tr("(exclude)");
+            txt += " " + tr("(排除)");
         return txt;
     }
 
diff --git a/src/filterdialog.ui b/src/filterdialog.ui
index aa81ec3..e351852 100644
--- a/src/filterdialog.ui
+++ b/src/filterdialog.ui
@@ -11,7 +11,7 @@
    </rect>
   </property>
   <property name="windowTitle">
-   <string>New Filter</string>
+   <string>新建过滤器</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
@@ -19,7 +19,7 @@
      <item row="0" column="0">
       <widget class="QLabel" name="labelMC">
        <property name="text">
-        <string>MC: 1.X</string>
+        <string>Minecraft: 1.X</string>
        </property>
       </widget>
      </item>
@@ -42,12 +42,12 @@
        </property>
        <item>
         <property name="text">
-         <string>Select category</string>
+         <string>选择类别</string>
         </property>
        </item>
        <item>
         <property name="text">
-         <string>Algorithm helpers</string>
+         <string>算法辅助点</string>
         </property>
         <property name="icon">
          <iconset resource="../rc/icons.qrc">
@@ -56,7 +56,7 @@
        </item>
        <item>
         <property name="text">
-         <string>Quad-structure</string>
+         <string>四联结构</string>
         </property>
         <property name="icon">
          <iconset resource="../rc/icons.qrc">
@@ -65,7 +65,7 @@
        </item>
        <item>
         <property name="text">
-         <string>Structures</string>
+         <string>结构</string>
         </property>
         <property name="icon">
          <iconset resource="../rc/icons.qrc">
@@ -74,7 +74,7 @@
        </item>
        <item>
         <property name="text">
-         <string>Biome filter</string>
+         <string>主世界生物群系</string>
         </property>
         <property name="icon">
          <iconset resource="../rc/icons.qrc">
@@ -83,7 +83,7 @@
        </item>
        <item>
         <property name="text">
-         <string>Nether biomes</string>
+         <string>下界生物群系</string>
         </property>
         <property name="icon">
          <iconset resource="../rc/icons.qrc">
@@ -92,7 +92,7 @@
        </item>
        <item>
         <property name="text">
-         <string>End biomes</string>
+         <string>末地生物群系</string>
         </property>
         <property name="icon">
          <iconset resource="../rc/icons.qrc">
@@ -101,7 +101,7 @@
        </item>
        <item>
         <property name="text">
-         <string>Others</string>
+         <string>其他</string>
         </property>
         <property name="icon">
          <iconset resource="../rc/icons.qrc">
@@ -135,7 +135,7 @@
        </property>
        <item>
         <property name="text">
-         <string>Select filter</string>
+         <string>选择过滤器</string>
         </property>
        </item>
       </widget>
@@ -143,7 +143,7 @@
      <item row="0" column="1">
       <widget class="QLabel" name="labelFilterType">
        <property name="text">
-        <string>Filter type:</string>
+        <string>过滤器类型:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
@@ -163,7 +163,7 @@
        <item>
         <widget class="QGroupBox" name="groupBoxPosition">
          <property name="title">
-          <string>Location</string>
+          <string>位置</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="1" column="3">
@@ -193,7 +193,7 @@
            <widget class="QComboBox" name="comboBoxRelative">
             <item>
              <property name="text">
-              <string>World origin</string>
+              <string>世界原点</string>
              </property>
             </item>
            </widget>
@@ -208,7 +208,7 @@
           <item row="3" column="0">
            <widget class="QRadioButton" name="radioCustom">
             <property name="text">
-             <string>Custom:</string>
+             <string>自定义区域:</string>
             </property>
            </widget>
           </item>
@@ -217,7 +217,7 @@
             <item>
              <widget class="QLabel" name="labelX1">
               <property name="toolTip">
-               <string>Lower bound (inclusive)</string>
+               <string>最小值 (包含)</string>
               </property>
               <property name="text">
                <string>X&lt;sub&gt;1&lt;/sub&gt;:</string>
@@ -227,7 +227,7 @@
             <item>
              <widget class="QLineEdit" name="lineEditX1">
               <property name="toolTip">
-               <string>Lower bound (inclusive)</string>
+               <string>最小值 (包含)</string>
               </property>
               <property name="text">
                <string>0</string>
@@ -237,7 +237,7 @@
             <item>
              <widget class="QLabel" name="labelZ1">
               <property name="toolTip">
-               <string>Lower bound (inclusive)</string>
+               <string>最小值 (包含)</string>
               </property>
               <property name="text">
                <string>Z&lt;sub&gt;1&lt;/sub&gt;:</string>
@@ -247,7 +247,7 @@
             <item>
              <widget class="QLineEdit" name="lineEditZ1">
               <property name="toolTip">
-               <string>Lower bound (inclusive)</string>
+               <string>最小值 (包含)</string>
               </property>
               <property name="text">
                <string>0</string>
@@ -270,7 +270,7 @@
             <item>
              <widget class="QLabel" name="labelX2">
               <property name="toolTip">
-               <string>Upper bound (inclusive)</string>
+               <string>最大值 (包含)</string>
               </property>
               <property name="text">
                <string>X&lt;sub&gt;2&lt;/sub&gt;:</string>
@@ -280,7 +280,7 @@
             <item>
              <widget class="QLineEdit" name="lineEditX2">
               <property name="toolTip">
-               <string>Upper bound (inclusive)</string>
+               <string>最大值 (包含)</string>
               </property>
               <property name="text">
                <string>0</string>
@@ -290,7 +290,7 @@
             <item>
              <widget class="QLabel" name="labelZ2">
               <property name="toolTip">
-               <string>Upper bound (inclusive)</string>
+               <string>最大值 (包含)</string>
               </property>
               <property name="text">
                <string>Z&lt;sub&gt;2&lt;/sub&gt;:</string>
@@ -300,7 +300,7 @@
             <item>
              <widget class="QLineEdit" name="lineEditZ2">
               <property name="toolTip">
-               <string>Upper bound (inclusive)</string>
+               <string>最大值 (包含)</string>
               </property>
               <property name="text">
                <string>0</string>
@@ -312,7 +312,7 @@
           <item row="2" column="0" colspan="3">
            <widget class="QRadioButton" name="radioSquare">
             <property name="text">
-             <string>Within centred square of side:</string>
+             <string>设置正方形区域的边长:</string>
             </property>
             <property name="checked">
              <bool>true</bool>
@@ -322,21 +322,21 @@
           <item row="1" column="0" colspan="3">
            <widget class="QCheckBox" name="checkRadius">
             <property name="text">
-             <string>Within radial distance: </string>
+             <string>Within radial distance:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="3">
            <widget class="QLabel" name="labelSpinBox">
             <property name="text">
-             <string>Instances within area:</string>
+             <string>区域内的实例数:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0" colspan="3">
            <widget class="QLabel" name="labelRelative">
             <property name="text">
-             <string>Location is relative to:</string>
+             <string>中心位置相对于:</string>
             </property>
            </widget>
           </item>
@@ -353,7 +353,7 @@
          </property>
          <widget class="QWidget" name="tabBiomes">
           <attribute name="title">
-           <string>Biomes</string>
+           <string>生物群系</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="1" column="3">
@@ -372,14 +372,14 @@
               <string>Enables optimizations that trade accuracy for speed</string>
              </property>
              <property name="text">
-              <string>Approximate</string>
+              <string>近似搜索</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="labelY">
              <property name="text">
-              <string>Sample at height (Y):</string>
+              <string>采样高度 (Y):</string>
              </property>
             </widget>
            </item>
@@ -413,21 +413,21 @@
              <item>
               <widget class="QPushButton" name="buttonUncheck">
                <property name="text">
-                <string>uncheck all</string>
+                <string>清空全部</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="buttonInclude">
                <property name="text">
-                <string>include all</string>
+                <string>全部包含</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="buttonExclude">
                <property name="text">
-                <string>exclude all</string>
+                <string>全部排除</string>
                </property>
               </widget>
              </item>
@@ -450,13 +450,13 @@
          </widget>
          <widget class="QWidget" name="tabTemps">
           <attribute name="title">
-           <string>Temperature Categories</string>
+           <string>温度类型</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
-              <string>Select how many entries of each temperature category are required.</string>
+              <string>选择每个温度类型下需要的生物群系数目。</string>
              </property>
             </widget>
            </item>
@@ -480,13 +480,13 @@
          </widget>
          <widget class="QWidget" name="tabVariants">
           <attribute name="title">
-           <string>Variant</string>
+           <string>结构变体</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="1">
             <widget class="QCheckBox" name="checkAbandoned">
              <property name="text">
-              <string>Abandoned</string>
+              <string>包含废弃结构</string>
              </property>
             </widget>
            </item>
@@ -496,7 +496,7 @@
               <string>Allow only structures that start with one of the selected pieces</string>
              </property>
              <property name="text">
-              <string>Filter starting piece</string>
+              <string>仅搜索特定结构</string>
              </property>
             </widget>
            </item>
@@ -548,7 +548,7 @@
      <item>
       <widget class="QPushButton" name="buttonCancel">
        <property name="text">
-        <string>Cancel</string>
+        <string>取消</string>
        </property>
        <property name="icon">
         <iconset theme="dialog-cancel">
@@ -559,7 +559,7 @@
      <item>
       <widget class="QPushButton" name="buttonOk">
        <property name="text">
-        <string>Ok</string>
+        <string>确定</string>
        </property>
        <property name="icon">
         <iconset theme="dialog-ok">
-- 
2.35.1

