From d1ae2d70dec0581f6c5f3edf95b34bc5d286d7cb Mon Sep 17 00:00:00 2001
From: The jR3f <97492766+jR3f@users.noreply.github.com>
Date: Tue, 11 Jan 2022 16:23:41 +0800
Subject: [PATCH] Translate search to zh-CHS


diff --git a/src/search.h b/src/search.h
index a9bddf2..88039e1 100644
--- a/src/search.h
+++ b/src/search.h
@@ -132,13 +132,13 @@ static const struct FilterList
         list[F_SCALE_TO_NETHER] = FilterInfo{
             CAT_HELPER, 0, 0, 0, 0, 0, 0, 1, 0, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/portal.png",
-            _("Coordinate factor x/8"),
+            _("坐标系数 x/8"),
             _("Divides relative location by 8, from Overworld to Nether.")
         };
         list[F_SCALE_TO_OVERWORLD] = FilterInfo{
             CAT_HELPER, 0, 0, 0, 0, 0, 0, 1, 0, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/portal.png",
-            _("Coordinate factor x*8"),
+            _("坐标系数 x*8"),
             _("Multiplies relative location by 8, from Nether to Overworld.")
         };
         const char *ref_desc = _(
@@ -150,50 +150,50 @@ static const struct FilterList
         list[F_REFERENCE_1] = FilterInfo{
             CAT_HELPER, 0, 1, 1, 0, 0, 0, 1, 0, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/reference.png",
-            _("Reference point 1:1"),
+            _("参考点 1:1"),
             ref_desc
         };
         list[F_REFERENCE_4] = FilterInfo{
             CAT_HELPER, 0, 1, 1, 0, 0, 0, 4, 0, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/reference.png",
-            _("Reference point 1:4"),
+            _("参考点 1:4"),
             ref_desc
         };
         list[F_REFERENCE_16] = FilterInfo{
             CAT_HELPER, 0, 1, 1, 0, 0, 0, 16, 0, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/reference.png",
-            _("Reference point 1:16"),
+            _("参考点 1:16"),
             ref_desc
         };
         list[F_REFERENCE_64] = FilterInfo{
             CAT_HELPER, 0, 1, 1, 0, 0, 0, 64, 0, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/reference.png",
-            _("Reference point 1:64"),
+            _("参考点 1:64"),
             ref_desc
         };
         list[F_REFERENCE_256] = FilterInfo{
             CAT_HELPER, 0, 1, 1, 0, 0, 0, 256, 0, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/reference.png",
-            _("Reference point 1:256"),
+            _("参考点 1:256"),
             ref_desc
         };
         list[F_REFERENCE_512] = FilterInfo{
             CAT_HELPER, 0, 1, 1, 0, 0, 0, 512, 0, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/reference.png",
-            _("Reference point 1:512"),
+            _("参考点 1:512"),
             ref_desc
         };
         list[F_REFERENCE_1024] = FilterInfo{
             CAT_HELPER, 0, 1, 1, 0, 0, 0, 1024, 0, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/reference.png",
-            _("Reference point 1:1024"),
+            _("参考点 1:1024"),
             ref_desc
         };
 
         list[F_QH_IDEAL] = FilterInfo{
             CAT_QUAD, 0, 1, 1, 0, 0, Swamp_Hut, 512, 0, MC_1_4, MC_NEWEST, 0, 0, disp++,
             ":icons/quad.png",
-            _("Quad-hut (ideal)"),
+            _("四联女巫小屋 (理想)"),
             _("The lower 48-bits provide potential for four swamp huts in "
             "spawning range, in one of the best configurations that exist.")
         };
@@ -201,7 +201,7 @@ static const struct FilterList
         list[F_QH_CLASSIC] = FilterInfo{
             CAT_QUAD, 0, 1, 1, 0, 0, Swamp_Hut, 512, 0, MC_1_4, MC_NEWEST, 0, 0, disp++,
             ":icons/quad.png",
-            _("Quad-hut (classic)"),
+            _("四联女巫小屋 (经典)"),
             _("The lower 48-bits provide potential for four swamp huts in "
             "spawning range, in one of the \"classic\" configurations. "
             "(Checks for huts in the nearest 2x2 chunk corners of each "
@@ -211,7 +211,7 @@ static const struct FilterList
         list[F_QH_NORMAL] = FilterInfo{
             CAT_QUAD, 0, 1, 1, 0, 0, Swamp_Hut, 512, 0, MC_1_4, MC_NEWEST, 0, 0, disp++,
             ":icons/quad.png",
-            _("Quad-hut (normal)"),
+            _("四联女巫小屋 (普通)"),
             _("The lower 48-bits provide potential for four swamp huts in "
             "spawning range, such that all of them are within 128 blocks "
             "of a single AFK location, including a vertical tollerance "
@@ -221,7 +221,7 @@ static const struct FilterList
         list[F_QH_BARELY] = FilterInfo{
             CAT_QUAD, 0, 1, 1, 0, 0, Swamp_Hut, 512, 0, MC_1_4, MC_NEWEST, 0, 0, disp++,
             ":icons/quad.png",
-            _("Quad-hut (barely)"),
+            _("四联女巫小屋 (稀疏)"),
             _("The lower 48-bits provide potential for four swamp huts in "
             "spawning range, in any configuration, such that the bounding "
             "boxes are within 128 blocks of a single AFK location.")
@@ -230,7 +230,7 @@ static const struct FilterList
         list[F_QM_95] = FilterInfo{
             CAT_QUAD, 0, 1, 1, 0, 0, Monument, 512, 0, MC_1_8, MC_NEWEST, 0, 0, disp++,
             ":icons/quad.png",
-            _("Quad-ocean-monument (>95%)"),
+            _("四联海底神殿 (>95%)"),
             _("The lower 48-bits provide potential for 95% of the area of "
             "four ocean monuments to be within 128 blocks of an AFK "
             "location.")
@@ -239,7 +239,7 @@ static const struct FilterList
         list[F_QM_90] = FilterInfo{
             CAT_QUAD, 0, 1, 1, 0, 0, Monument, 512, 0, MC_1_8, MC_NEWEST, 0, 0, disp++,
             ":icons/quad.png",
-            _("Quad-ocean-monument (>90%)"),
+            _("四联海底神殿 (>90%)"),
             _("The lower 48-bits provide potential for 90% of the area of "
             "four ocean monuments to be within 128 blocks of an AFK "
             "location.")
@@ -248,7 +248,7 @@ static const struct FilterList
         list[F_BIOME] = FilterInfo{
             CAT_BIOMES, 1, 1, 1, 0, L_VORONOI_1, 0, 1, 0, MC_1_0, MC_1_17, 0, 1, disp++, // disable for 1.18
             ":icons/map.png",
-            _("Biome filter 1:1"),
+            _("主世界生物群系 1:1"),
             _("Only seeds with the included (+) biomes in the specified area and "
             "discard those that have biomes that are explicitly excluded (-).")
         };
@@ -256,28 +256,28 @@ static const struct FilterList
         list[F_BIOME_4] = FilterInfo{
             CAT_BIOMES, 1, 1, 1, 0, 0, 0, 4, 0, MC_1_0, MC_NEWEST, 0, 1, disp++,
             ":icons/map.png",
-            _("Biome filter 1:4"),
+            _("主世界生物群系 1:4"),
             _("Only seeds with the included (+) biomes in the specified area and "
             "discard those that have biomes that are explicitly excluded (-).")
         };
         list[F_BIOME_16] = FilterInfo{
             CAT_BIOMES, 1, 1, 1, 0, 0, 0, 16, 0, MC_1_0, MC_NEWEST, 0, 1, disp++,
             ":icons/map.png",
-            _("Biome filter 1:16"),
+            _("主世界生物群系 1:16"),
             _("Only seeds with the included (+) biomes in the specified area and "
             "discard those that have biomes that are explicitly excluded (-).")
         };
         list[F_BIOME_64] = FilterInfo{
             CAT_BIOMES, 1, 1, 1, 0, 0, 0, 64, 0, MC_1_0, MC_NEWEST, 0, 1, disp++,
             ":icons/map.png",
-            _("Biome filter 1:64"),
+            _("主世界生物群系 1:64"),
             _("Only seeds with the included (+) biomes in the specified area and "
             "discard those that have biomes that are explicitly excluded (-).")
         };
         list[F_BIOME_256] = FilterInfo{
             CAT_BIOMES, 1, 1, 1, 0, 0, 0, 256, 0, MC_1_0, MC_NEWEST, 0, 1, disp++,
             ":icons/map.png",
-            _("Biome filter 1:256"),
+            _("主世界生物群系 1:256"),
             _("Only seeds with the included (+) biomes in the specified area and "
             "discard those that have biomes that are explicitly excluded (-).")
         };
@@ -285,7 +285,7 @@ static const struct FilterList
         list[F_BIOME_4_RIVER] = FilterInfo{
             CAT_BIOMES, 1, 1, 1, 0, L_RIVER_MIX_4, 0, 4, 0, MC_1_13, MC_1_17, 0, 0, disp++,
             ":icons/map.png",
-            _("Biome filter 1:4 RIVER"),
+            _("主世界生物群系 1:4 RIVER"),
             _("Only seeds with the included (+) biomes in the specified area and "
             "discard those that have biomes that are explicitly excluded (-) "
             "at layer RIVER with scale 1:4. "
@@ -294,7 +294,7 @@ static const struct FilterList
         list[F_BIOME_256_OTEMP] = FilterInfo{
             CAT_BIOMES, 0, 1, 1, 0, L_OCEAN_TEMP_256, 0, 256, 0, MC_1_13, MC_1_17, 0, 0, disp++,
             ":icons/map.png",
-            _("Biome filter 1:256 O.TEMP"),
+            _("主世界生物群系 1:256 O.TEMP"),
             _("Only seeds with the included (+) biomes in the specified area and "
             "discard those that have biomes that are explicitly excluded (-) "
             "at layer OCEAN TEMPERATURE with scale 1:256. "
@@ -303,77 +303,77 @@ static const struct FilterList
         list[F_TEMPS] = FilterInfo{
             CAT_BIOMES, 1, 1, 1, 0, 0, 0, 1024, 0, MC_1_7, MC_1_17, 0, 0, disp++,
             ":icons/tempcat.png",
-            _("Temperature categories"),
+            _("温度类别"),
             _("Checks that the area has a minimum of all the required temperature categories.")
         };
 
         list[F_BIOME_NETHER_1] = FilterInfo{
             CAT_NETHER, 1, 1, 1, 0, 0, 0, 1, 0, MC_1_16, 0, -1, 1, disp++, // disabled
             ":icons/nether.png",
-            _("Nether biome filter 1:1 (disabled)"),
+            _("下界生物群系 1:1 (禁用)"),
             _("Nether biomes after voronoi scaling to 1:1.")
         };
         list[F_BIOME_NETHER_4] = FilterInfo{
             CAT_NETHER, 0, 1, 1, 0, 0, 0, 4, 0, MC_1_16, MC_NEWEST, -1, 0, disp++,
             ":icons/nether.png",
-            _("Nether biome filter 1:4"),
+            _("下界生物群系 1:4"),
             _("Nether biomes with normal noise sampling at scale 1:4.")
         };
         list[F_BIOME_NETHER_16] = FilterInfo{
             CAT_NETHER, 0, 1, 1, 0, 0, 0, 16, 0, MC_1_16, MC_NEWEST, -1, 0, disp++,
             ":icons/nether.png",
-            _("Nether biome filter 1:16"),
+            _("下界生物群系 1:16"),
             _("Nether biomes, but only sampled at scale 1:16.")
         };
         list[F_BIOME_NETHER_64] = FilterInfo{
             CAT_NETHER, 0, 1, 1, 0, 0, 0, 64, 0, MC_1_16, MC_NEWEST, -1, 0, disp++,
             ":icons/nether.png",
-            _("Nether biome filter 1:64"),
+            _("下界生物群系 1:64"),
             _("Nether biomes, but only sampled at scale 1:64.")
         };
         list[F_BIOME_NETHER_256] = FilterInfo{
             CAT_NETHER, 0, 1, 1, 0, 0, 0, 256, 0, MC_1_16, MC_NEWEST, -1, 0, disp++,
             ":icons/nether.png",
-            _("Nether biome filter 1:256"),
+            _("下界生物群系 1:256"),
             _("Nether biomes, but only sampled at scale 1:256.")
         };
 
         list[F_BIOME_END_1] = FilterInfo{
             CAT_END, 1, 1, 1, 0, 0, 0, 1, 0, MC_1_9, 0, +1, 1, disp++, // disabled
             ":icons/the_end.png",
-            _("End biome filter 1:1 (disabled)"),
+            _("末地生物群系 1:1 (禁用)"),
             _("End biomes after voronoi scaling to 1:1.")
         };
         list[F_BIOME_END_4] = FilterInfo{
             CAT_END, 0, 1, 1, 0, 0, 0, 4, 0, MC_1_9, MC_NEWEST, +1, 0, disp++,
             ":icons/the_end.png",
-            _("End biome filter 1:4"),
+            _("末地生物群系 1:4"),
             _("End biomes sampled at scale 1:4. Note this is just a simple upscale of 1:16.")
         };
         list[F_BIOME_END_16] = FilterInfo{
             CAT_END, 0, 1, 1, 0, 0, 0, 16, 0, MC_1_9, MC_NEWEST, +1, 0, disp++,
             ":icons/the_end.png",
-            _("End biome filter 1:16"),
+            _("末地生物群系 1:16"),
             _("End biomes with normal sampling at scale 1:16. ")
         };
         list[F_BIOME_END_64] = FilterInfo{
             CAT_END, 0, 1, 1, 0, 0, 0, 64, 0, MC_1_9, MC_NEWEST, +1, 0, disp++,
             ":icons/the_end.png",
-            _("End biome filter 1:64"),
+            _("末地生物群系 1:64"),
             _("End biomes with lossy sampling at scale 1:64. ")
         };
 
         list[F_SPAWN] = FilterInfo{
             CAT_OTHER, 1, 1, 1, 1, 0, 0, 1, 0, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/spawn.png",
-            _("Spawn"),
+            _("出生点"),
             ""
         };
 
         list[F_SLIME] = FilterInfo{
             CAT_OTHER, 0, 1, 1, 0, 0, 0, 16, 1, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/slime.png",
-            _("Slime chunk"),
+            _("史莱姆区块"),
             ""
         };
 
@@ -388,28 +388,28 @@ static const struct FilterList
         list[F_STRONGHOLD] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, 0, 1, 1, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/stronghold.png",
-            _("Stronghold"),
+            _("要塞"),
             ""
         };
 
         list[F_VILLAGE] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Village, 1, 1, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/village.png",
-            _("Village"),
+            _("村庄"),
             ""
         };
 
         list[F_MINESHAFT] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 0, 0, Mineshaft, 1, 1, MC_1_0, MC_NEWEST, 0, 0, disp++,
             ":icons/mineshaft.png",
-            _("Abandoned mineshaft"),
+            _("废弃矿井"),
             ""
         };
 
         list[F_DESERT] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Desert_Pyramid, 1, 1, MC_1_3, MC_NEWEST, 0, 0, disp++,
             ":icons/desert.png",
-            _("Desert pyramid"),
+            _("沙漠神殿"),
             _("In version 1.18, desert pyramids depend on surface height and may fail to "
             "generate near caves/aquifers, rivers and oceans.")
         };
@@ -417,7 +417,7 @@ static const struct FilterList
         list[F_JUNGLE] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Jungle_Temple, 1, 1, MC_1_3, MC_NEWEST, 0, 0, disp++,
             ":icons/jungle.png",
-            _("Jungle temple"),
+            _("丛林神庙"),
             _("In version 1.18, jungle temples depend on surface height and may fail to "
             "generate near caves/aquifers, rivers and oceans.")
         };
@@ -425,28 +425,28 @@ static const struct FilterList
         list[F_HUT] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Swamp_Hut, 1, 1, MC_1_4, MC_NEWEST, 0, 0, disp++,
             ":icons/hut.png",
-            _("Swamp hut"),
+            _("女巫小屋"),
             ""
         };
 
         list[F_MONUMENT] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Monument, 1, 1, MC_1_8, MC_NEWEST, 0, 0, disp++,
             ":icons/monument.png",
-            _("Ocean monument"),
+            _("海底神殿"),
             ""
         };
 
         list[F_IGLOO] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Igloo, 1, 1, MC_1_9, MC_NEWEST, 0, 0, disp++,
             ":icons/igloo.png",
-            _("Igloo"),
+            _("雪屋"),
             ""
         };
 
         list[F_MANSION] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Mansion, 1, 1, MC_1_11, MC_NEWEST, 0, 0, disp++,
             ":icons/mansion.png",
-            _("Woodland mansion"),
+            _("林地府邸"),
             _("In version 1.18, mansions depend on surface height and may fail to "
             "generate near caves/aquifers, rivers and oceans.")
         };
@@ -454,21 +454,21 @@ static const struct FilterList
         list[F_RUINS] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Ocean_Ruin, 1, 1, MC_1_13, MC_NEWEST, 0, 0, disp++,
             ":icons/ruins.png",
-            _("Ocean ruins"),
+            _("海底废墟"),
             ""
         };
 
         list[F_SHIPWRECK] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Shipwreck, 1, 1, MC_1_13, MC_NEWEST, 0, 0, disp++,
             ":icons/shipwreck.png",
-            _("Shipwreck"),
+            _("沉船"),
             ""
         };
 
         list[F_TREASURE] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Treasure, 1, 1, MC_1_13, MC_NEWEST, 0, 0, disp++,
             ":icons/treasure.png",
-            _("Buried treasure"),
+            _("埋藏的宝藏"),
             _("Buried treasures are always positioned near the center of a chunk "
             "rather than a chunk boarder. Make sure the testing area is set "
             "accordingly.")
@@ -477,49 +477,49 @@ static const struct FilterList
         list[F_OUTPOST] = FilterInfo{
             CAT_STRUCT, 1, 1, 1, 1, 0, Outpost, 1, 1, MC_1_14, MC_NEWEST, 0, 0, disp++,
             ":icons/outpost.png",
-            _("Pillager outpost"),
+            _("掠夺者前哨站"),
             ""
         };
 
         list[F_PORTAL] = FilterInfo{
             CAT_STRUCT, 0, 1, 1, 1, 0, Ruined_Portal, 1, 1, MC_1_16, MC_NEWEST, 0, 0, disp++,
             ":icons/portal.png",
-            _("Ruined portal (overworld)"),
+            _("废弃传送门 (主世界)"),
             ""
         };
 
         list[F_PORTALN] = FilterInfo{
             CAT_STRUCT, 0, 1, 1, 1, 0, Ruined_Portal_N, 1, 1, MC_1_16, MC_NEWEST, -1, 0, disp++,
             ":icons/portal.png",
-            _("Ruined portal (nether)"),
+            _("废弃传送门 (下界)"),
             ""
         };
 
         list[F_FORTRESS] = FilterInfo{
             CAT_STRUCT, 0, 1, 1, 1, 0, Fortress, 1, 1, MC_1_0, MC_NEWEST, -1, 0, disp++,
             ":icons/fortress.png",
-            _("Nether fortress"),
+            _("下界要塞"),
             ""
         };
 
         list[F_BASTION] = FilterInfo{
             CAT_STRUCT, 0, 1, 1, 1, 0, Bastion, 1, 1, MC_1_16, MC_NEWEST, -1, 0, disp++,
             ":icons/bastion.png",
-            _("Bastion remnant"),
+            _("堡垒遗迹"),
             ""
         };
 
         list[F_ENDCITY] = FilterInfo{
             CAT_STRUCT, 0, 1, 1, 1, 0, End_City, 1, 1, MC_1_9, MC_NEWEST, +1, 0, disp++,
             ":icons/endcity.png",
-            _("End city"),
+            _("末地城"),
             ""
         };
 
         list[F_GATEWAY] = FilterInfo{
             CAT_STRUCT, 0, 1, 1, 1, 0, End_Gateway, 1, 1, MC_1_13, MC_NEWEST, +1, 0, disp++,
             ":icons/gateway.png",
-            _("End gateway"),
+            _("末地折跃门"),
             _("Scattered end gateway return portals, not including those "
             "generated when defeating the dragon.")
         };
-- 
2.35.1

